### PR TITLE
quic: fix stalled virtual listener

### PR DESCRIPTION
**Pull request by MarcoPolo**
Fixes a case that only really shows up in CI where a listener can be stalled. This lead to some test flakiness.

### CODE REVIEW

**Comment by marten-seemann** [comment]
How is this different from #2029?

File: p2p/transport/quicreuse/reuse.go [hunk 1]
```diff
@@ -110,7 +110,8 @@ func (r *reuse) gc() {
 		select {
 		case <-r.closeChan:
 			return
-		case now := <-ticker.C:
+		case <-ticker.C:
+			now := time.Now()
```
Comment (marten-seemann) [inline p2p/transport/quicreuse/reuse.go:114]: This seems unrelated?

**Review by marten-seemann — APPROVED** [review]

File: p2p/transport/quic/virtuallistener.go [hunk 2]
```diff
@@ -146,9 +146,19 @@ func (r *acceptLoopRunner) innerAccept(l *listener, expectedVersion quic.Version
...
 	for {
-		r.acceptSem <- struct{}{}
-		conn, err := r.innerAccept(l, expectedVersion, bufferedConnChan)
-		<-r.acceptSem
+		var conn tpt.CapableConn
+		var err error
+		select {
+		case r.acceptSem <- struct{}{}:
+			conn, err = r.innerAccept(l, expectedVersion, bufferedConnChan)
+			<-r.acceptSem
```
Comment (marten-seemann) [inline p2p/transport/quic/virtuallistener.go:154]: Should the `conn == nil && err == nil` be moved into this `case` for clarity?

**Review by marten-seemann — APPROVED** [review]

File: p2p/transport/quicreuse/reuse.go (hunk 1 above)
Comment (MarcoPolo) [reply p2p/transport/quicreuse/reuse.go:114]: I'll back this out
